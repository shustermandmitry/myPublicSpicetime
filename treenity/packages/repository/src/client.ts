export * from './client/access-token';
export * from './client/create-feathers-client';
export * from './client/simple-client';
export * from './utils/cookies';
export * from './client/socket-custom-methods';
export * from './mods/client';
export * from './client/parse-jwt';
export * from './utils/email-parse';
export { makeServiceActions } from '@/mods/entity/service-entity-actions';
