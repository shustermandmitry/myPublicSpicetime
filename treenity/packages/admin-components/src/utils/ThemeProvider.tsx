/*
 * Copyright (c) 2024. Treenity Inc.
 */

import { setCookie, useCookie } from '@treenity/repository/client';
import { TreeNode, useTree } from '@treenity/tree-api';
import { useStorage } from '@treenity/ui-kit/hooks';
import { Theme, ThemeEntity, ThemeProvider } from '@treenity/ui-kit/theme';
import React, { FC, PropsWithChildren } from 'react';
import useSWR from 'swr';

const defaultThemeName = 'light';

function useLoadTheme(name: string): Theme | undefined {
  const treeApi = useTree();
  const { data: themeNode } = treeApi.useNode(`/sys/themes/${name}`);
  const { data } = useSWR(`${name}-theme` + (themeNode as any)?.$id, () =>
    (themeNode as TreeNode)?.get(ThemeEntity),
  );
  return data;
}

const ThemeProviderLoader: FC<PropsWithChildren> = ({ children }) => {
  const lightTheme = useLoadTheme('light');
  const darkTheme = useLoadTheme('dark');

  const cookies = useCookie();
  const [storedThemeName, _setThemeName] = useStorage<string>(
    'theme',
    cookies.theme || defaultThemeName,
  );
  const setThemeName = (theme: string) => {
    setCookie('theme', theme, 365 * 24 * 60 * 60);
    _setThemeName(theme);
  };

  if (!lightTheme || !darkTheme) return null;

  return (
    <ThemeProvider
      light={lightTheme}
      dark={darkTheme}
      children={children}
      storedThemeName={storedThemeName}
      setThemeName={setThemeName}
    />
  );
};

export default ThemeProviderLoader;
