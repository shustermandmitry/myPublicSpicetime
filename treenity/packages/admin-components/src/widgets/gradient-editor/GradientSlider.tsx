/*
 * Copyright (c) 2024. Treenity Inc.
 */

import styled from '@emotion/styled';
import { omitProps } from '@treenity/ui-kit/utils'; // убедитесь, что эта функция правильно импортирована
import { Slider } from 'antd';
import { SliderRangeProps } from 'antd/es/slider';
import React, { FC, forwardRef } from 'react';

interface SliderRef {
  focus(): void;
  blur(): void;
}

export interface GradientSliderColors {
  color: string | null;
  position: number;
}

interface GradientSliderProps {
  colors: GradientSliderColors[];
}

type GradientSliderType = SliderRangeProps & React.RefAttributes<SliderRef> & GradientSliderProps;

const GradientSlider: FC<GradientSliderType> = forwardRef<SliderRef, GradientSliderType>(
  ({ ...restProps }, ref) => {
    return <SliderStyled ref={ref} {...restProps} />;
  },
);

const SliderStyled = styled(Slider, omitProps('colors'))<{
  colors: GradientSliderColors[];
}>`
  .ant-slider-rail {
    background: linear-gradient(
      ${p => `90deg, ${p.colors.map(({ color, position }, index) => `${color} ${position}%`)}`}
    );
  }

  .ant-slider-track {
    background-color: transparent !important;
  }
`;

export default GradientSlider;
