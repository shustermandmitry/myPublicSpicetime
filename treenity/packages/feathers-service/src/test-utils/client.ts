import {
  Application,
  defaultEventMap,
  defaultServiceMethods,
  RealTimeConnection,
  TransportConnection,
} from '@feathersjs/feathers';
import { Service, SocketService } from '@feathersjs/transport-commons/client';
import { runMethod } from '@feathersjs/transport-commons/lib/socket/utils';

export { SocketService };

export default function testClient<Services = any>(options?: any) {
  const defaultService = function (this: Application<any>, name: string) {
    const self = this;
    const events = Object.values(defaultEventMap);
    const connection = {
      emit(method: string, ...args: any[]) {
        return runMethod(self, {} as RealTimeConnection, name, method, args);
      },
    };

    const settings = Object.assign({}, options, {
      events,
      name,
      connection,
      method: 'emit',
    });

    return new Service(settings) as any;
  };

  const initialize = function (app: Application<Services>) {
    app.defaultService = defaultService;
    app.mixins.unshift((service, _location, options) => {
      if (options && options.methods && service instanceof Service) {
        const customMethods = options.methods.filter(name => !defaultServiceMethods.includes(name));

        service.methods(...customMethods);
      }
    });
  };

  initialize.Service = Service;
  initialize.service = defaultService;

  return initialize as TransportConnection<Services>;
}

if (typeof module !== 'undefined') {
  module.exports = Object.assign(testClient, module.exports);
}
