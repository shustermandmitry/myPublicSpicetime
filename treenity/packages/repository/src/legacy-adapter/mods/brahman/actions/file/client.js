import './TelegramActionFileSend.client';
import './TelegramActionFileUpload.client';
import './TelegramFile.client';
