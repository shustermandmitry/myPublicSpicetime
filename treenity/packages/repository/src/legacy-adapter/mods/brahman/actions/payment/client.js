import './TelegramActionGetWallet.client';
import './TelegramActionSendCoins.client';
import './TelegramActionSplitCoins.client';
