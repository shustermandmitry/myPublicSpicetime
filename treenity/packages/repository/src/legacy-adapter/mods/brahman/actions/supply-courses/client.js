import './TelegramActionSupplyCourses.client';
import './TelegramCoursesOperator.client';
import './TelegramActionBecomeOperator.client';
