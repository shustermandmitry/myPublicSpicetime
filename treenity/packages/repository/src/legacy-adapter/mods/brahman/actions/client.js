import './addActionMenu';
import './back/TelegramActionBack.client';
import './broadcast/TelegramActionBroadcast.client';

import './call-command-by-param/TGActionCallCommandByParam.client';
import './crypto/TelegramActionGetRates.client';
// import './voice/video/SynhesizeAudioVideo';
import './emit-text/EmitText.client';
import './file/TelegramActionFileSend.client';
import './file/TelegramActionFileUpload.client';
import './file/TelegramFile.client';
import './forward/TelegramActionForward.client';
import './game/CurrentGame.client';
import './game/Game.client';
import './game/TelegramActionGame.client';
import './get-value/GetValue.client';
import './i18n/TGActionSelectLanguage.client';
import './if-else/TelegramActionIfElse.client';

import './keyword-select/TelegramActionKeywordSelect.client';
import './menu/TelegramActionMenu.client';
import './message/TelegramActionMessage.client';
import './on-error/TelegramActionOnError.client';
import './page/TelegramActionPage.client';
// import './pantaro/client';
import './params/TelegramActionParams.client';
import './payment/client';
import './question/TelegramActionQuestion.client';
import './removeMessage/TelegramActionRemove.client';
import './reset-history/ResetSession.client';
import './reset-session/ResetSession.client';
import './set-telegram-user/TGActionSetTelegramUser.client';
import './set-value/SetValue.client';

import './supply-courses/client';

import './tag/SetTag';
import './TelegramAction.client';
import './timers/TgActionSetTimer';

import './timers/timers.service.client';

import './translate/TString.client';
import './voice/tinkoff/TinkoffVoiceRecognition.client';
import './voice/tinkoff/TinkoffVoiceSynthesize.client';

import './voice/yandex/YandexVoiceRecognition.client';
import './voice/yandex/YandexVoiceSynthesize.client';

import './ym-statistics/TelegramPageStatistics.client';
