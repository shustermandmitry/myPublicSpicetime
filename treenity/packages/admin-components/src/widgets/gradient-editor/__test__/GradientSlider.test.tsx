/*
 * Copyright (c) 2024. Treenity Inc.
 */

import ThemeProvider from '@/utils/ThemeProvider';
import { jest } from '@jest/globals';
import { render } from '@testing-library/react';
import { createCookies, parseCookies, setCookie, CookieContext } from '@treenity/repository/client';
import GradientSlider, { GradientSliderColors } from '../GradientSlider';

const defaultValue: number[] = [0, 100];

const cookies = createCookies(parseCookies(document.cookie), setCookie);
let valueMock: number[] = defaultValue;
let colorsMock: GradientSliderColors[] = [
  { position: valueMock[0], color: 'red' },
  { position: valueMock[1], color: 'green' },
];

const updateMock = jest.fn((value: number[]) => {
  valueMock = value;
  return value;
});

describe('GradientSlider', () => {
  beforeEach(() => {
    updateMock.mockClear();
    valueMock = defaultValue;
  });

  it('render GradientSlider component', async () => {
    const { container } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <GradientSlider
            onChange={updateMock}
            value={valueMock}
            colors={colorsMock}
            range={true}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`div[class^="ant-slider css-uvzcrr"]`);
    const sliders = root!.querySelectorAll(`div[class^="ant-slider-handle"][role="slider"]`);
    const rail = root!.querySelector(`div[class="ant-slider-rail"]`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(rail).toBeInTheDocument();
    expect(sliders).toHaveLength(valueMock.length);
    expect(rail).toHaveStyle(
      `background: linear-gradient(90deg, ${colorsMock
        .map((color, position) => `${color} ${position}`)
        .join(',')})`,
    );

    [...sliders].forEach((slider, index) => {
      expect(slider).toBeInTheDocument();
      expect(slider).toHaveAttribute('aria-valuenow', `${valueMock[index]}`);
    });
  });

  it('change slider', async () => {
    const newValues = [30, 70];
    const newColors: GradientSliderColors[] = [
      { position: newValues[0], color: 'red' },
      { position: newValues[1], color: 'green' },
    ];

    const { container, rerender } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <GradientSlider
            onChange={updateMock}
            value={valueMock}
            colors={colorsMock}
            range={true}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`div[class^="ant-slider css-uvzcrr"]`);
    const sliders = root!.querySelectorAll(`div[class^="ant-slider-handle"][role="slider"]`);
    const rail = root!.querySelector(`div[class="ant-slider-rail"]`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(rail).toBeInTheDocument();
    expect(sliders).toHaveLength(valueMock.length);
    expect(rail).toHaveStyle(
      `background: linear-gradient(90deg, ${colorsMock
        .map((color, position) => `${color} ${position}`)
        .join(',')})`,
    );

    [...sliders].forEach((slider, index) => {
      expect(slider).toBeInTheDocument();
      expect(slider).toHaveAttribute('aria-valuenow', `${valueMock[index]}`);
      expect(slider).toHaveAttribute('aria-valuemax', '100');
    });

    rerender(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <GradientSlider onChange={updateMock} value={newValues} colors={newColors} range={true} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    [...sliders].forEach((slider, index) => {
      expect(slider).toBeInTheDocument();
      expect(slider).toHaveAttribute('aria-valuenow', `${newValues[index]}`);
      expect(slider).toHaveAttribute('aria-valuemax', '100');
    });

    expect(rail).toHaveStyle(
      `background: linear-gradient(90deg, ${newColors
        .map((color, position) => `${color} ${position}`)
        .join(',')})`,
    );
  });
});
