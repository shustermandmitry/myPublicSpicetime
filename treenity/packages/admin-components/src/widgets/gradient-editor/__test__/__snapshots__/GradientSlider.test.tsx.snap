// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientSlider change slider 1`] = `
.emotion-0 .ant-slider-rail {
  background: linear-gradient(
        90deg, red 0%,green 100%
      );
}

.emotion-0 .ant-slider-track {
  background-color: transparent!important;
}

<div
  class="ant-slider emotion-0 emotion-1 ant-slider-horizontal"
>
  <div
    class="ant-slider-rail"
  />
  <div
    class="ant-slider-track ant-slider-track-1"
    style="left: 0%; width: 100%;"
  />
  <div
    class="ant-slider-step"
  />
  <div
    aria-disabled="false"
    aria-orientation="horizontal"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="0"
    class="ant-slider-handle ant-slider-handle-1"
    role="slider"
    style="left: 0%; transform: translateX(-50%);"
    tabindex="0"
  />
  <div
    aria-disabled="false"
    aria-orientation="horizontal"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="100"
    class="ant-slider-handle ant-slider-handle-2"
    role="slider"
    style="left: 100%; transform: translateX(-50%);"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    class="ant-slider-handle"
    style="transform: translateX(-50%); pointer-events: none; visibility: hidden;"
  />
</div>
`;

exports[`GradientSlider render GradientSlider component 1`] = `
.emotion-0 .ant-slider-rail {
  background: linear-gradient(
        90deg, red 0%,green 100%
      );
}

.emotion-0 .ant-slider-track {
  background-color: transparent!important;
}

<div
  class="ant-slider emotion-0 emotion-1 ant-slider-horizontal"
>
  <div
    class="ant-slider-rail"
  />
  <div
    class="ant-slider-track ant-slider-track-1"
    style="left: 0%; width: 100%;"
  />
  <div
    class="ant-slider-step"
  />
  <div
    aria-disabled="false"
    aria-orientation="horizontal"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="0"
    class="ant-slider-handle ant-slider-handle-1"
    role="slider"
    style="left: 0%; transform: translateX(-50%);"
    tabindex="0"
  />
  <div
    aria-disabled="false"
    aria-orientation="horizontal"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="100"
    class="ant-slider-handle ant-slider-handle-2"
    role="slider"
    style="left: 100%; transform: translateX(-50%);"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    class="ant-slider-handle"
    style="transform: translateX(-50%); pointer-events: none; visibility: hidden;"
  />
</div>
`;
